listenbrainz/influx-writer/influx-writer.py

```diff
@@ -191,18 +196,32 @@ def write(self, listen_dicts):
             timestamps = {}
             for result in results.get_points(measurement=get_measurement_name(user_name)):
                 dt = datetime.strptime(result['time'] , "%Y-%m-%dT%H:%M:%SZ")
-                timestamps[int(dt.strftime('%s'))] = 1
+                timestamps[int(dt.strftime('%s'))] = result
 
             for listen in users[user_name]['listens']:
-                # Check to see if the timestamp is already in the DB
                 t = int(listen['listened_at'])
-                if t in timestamps:
-                    duplicate_count += 1
-                    continue
-
-                unique_count += 1
-                submit.append(Listen().from_json(listen))
-                unique.append(listen)
+                artist_msid    = listen['track_metadata']['additional_info']['artist_msid']
+                recording_msid = listen['recording_msid']
+
+                # This will check if timestamps in the range (t - TIMESTAMP_DUPLICATE_DIFF, t + TIMESTAMP_DUPLICATE_DIFF)
+                # exist in the database for the user, with the same artist msid and recording msid. If it does, we
+                # consider this listen to be a duplicate and skip it.
+                for delta in range(TIMESTAMP_DUPLICATE_DIFF):
+                    fuzzed = t + delta
+                    if fuzzed in timestamps:
+                        if artist_msid == timestamps[fuzzed]['artist_msid'] and recording_msid == timestamps[fuzzed]['recording_msid']:
+                            duplicate_count += 1
+                            break
+
+                    fuzzed = t - delta
```

Comment:
> creating deltas is left as an exercise to the reader

`deltas = [0] + [sign * val for val in range(1, 31) for sign in (+1, -1)]`

No code duplication now :)

>I do like the positive/negative sweep though, this means that we hit the 0 case first, instead of going from -limit -> +limit

Exactly why I did that :D